tests/system/python/e2e/test_e2e_kafka.py

```diff
@@ -189,7 +189,7 @@ def _verify_consumer_data_from_topic(self, conn):
         r = r.read().decode()
         jdoc = json.loads(r)
```

Comment:
Add `assert len(jdoc)` As its giving error on local setup / AWS  with indexError. so that we can **avoid the error** and let next stuff fail if no data found and investigate later.

Reply:
Below is the detailed O/P on my local setup and it passes
```
pytest -s -vv tests/system/python/e2e/test_e2e_kafka.py 
============================================================= test session starts =============================================================
platform linux -- Python 3.5.3, pytest-3.6.0, py-1.7.0, pluggy-0.6.0 -- /usr/local/bin/python3.5
cachedir: tests/system/python/.pytest_cache
rootdir: /home/foglamp/Development/FogLAMP/tests/system/python, inifile: pytest.ini
plugins: mock-1.10.0, cov-2.5.1, asyncio-0.8.0, allure-adaptor-1.7.10, aiohttp-0.3.0
collected 1 item                                                                                                                              

tests/system/python/TestE2EKafka Test that data is inserted in FogLAMP and sent to Kafka
            start_south_north: Fixture that starts FogLAMP with south and north instance
            Assertions:
                on endpoint GET /foglamp/asset
                on endpoint GET /foglamp/asset/<asset_name>
                data received from Kafka is same as data sent <- e2e/test_e2e_kafka.py FogLAMP killed.
This script will remove all data stored in the SQLite3 datafile '/home/foglamp/Development/FogLAMP/data/foglamp.db'
Enter YES if you want to continue: Starting FogLAMP v1.5.1......
FogLAMP started.
FogLAMP v1.5.1 running.
FogLAMP Uptime:  6 seconds.
FogLAMP records: 0 read, 0 sent, 0 purged.
FogLAMP does not require authentication.
=== FogLAMP services:
foglamp.services.core
=== FogLAMP tasks:
Cloning into '/tmp/foglamp-south-coap'...
remote: Enumerating objects: 3, done.
remote: Counting objects: 100% (3/3), done.
remote: Compressing objects: 100% (3/3), done.
remote: Total 262 (delta 0), reused 0 (delta 0), pack-reused 259
Receiving objects: 100% (262/262), 44.19 KiB | 0 bytes/s, done.
Resolving deltas: 100% (88/88), done.
Checking connectivity... done.
Collecting aiocoap==0.3 (from -r /tmp/foglamp-south-coap/python/requirements-coap.txt (line 1))
  Using cached https://files.pythonhosted.org/packages/9c/f6/d839e4b14258d76e74a39810829c13f8dd31de2bfe0915579b2a609d1bbe/aiocoap-0.3.tar.gz
Collecting cbor2==4.0.0 (from -r /tmp/foglamp-south-coap/python/requirements-coap.txt (line 2))
  Using cached https://files.pythonhosted.org/packages/fd/1a/7e6d84e786d543f271a2922736acb4a28ddc0de76fefe0cff1dc6aa1c3fb/cbor2-4.0.0-py2.py3-none-any.whl
Installing collected packages: aiocoap, cbor2
  Running setup.py install for aiocoap ... done
Successfully installed aiocoap-0.3 cbor2-4.0.0
You are using pip version 9.0.1, however version 19.0.3 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
coap plugin is installed.
[sudo] password for foglamp: 
Ign:1 http://dl.google.com/linux/chrome/deb stable InRelease
Hit:2 http://packages.microsoft.com/repos/vscode stable InRelease                                                                             
Hit:3 http://dl.google.com/linux/chrome/deb stable Release                                                                                    
Hit:4 http://archive.ubuntu.com/ubuntu xenial InRelease                                                                                       
Hit:5 http://archive.canonical.com/ubuntu xenial InRelease                                                                                    
Hit:6 http://ppa.launchpad.net/fkrull/deadsnakes/ubuntu xenial InRelease                                             
Hit:7 http://apt.piduino.org stretch InRelease                                                 
Get:8 http://archive.ubuntu.com/ubuntu xenial-updates InRelease [109 kB]                            
Hit:9 https://deb.nodesource.com/node_8.x xenial InRelease                                                    
Hit:10 http://ppa.launchpad.net/jonathonf/python-3.5/ubuntu xenial InRelease                                  
Get:12 http://archive.ubuntu.com/ubuntu xenial-backports InRelease [107 kB]                                   
Hit:13 http://ppa.launchpad.net/linuxgndu/sqlitebrowser/ubuntu xenial InRelease                     
Get:14 http://archive.ubuntu.com/ubuntu xenial-security InRelease [109 kB]                          
Hit:15 http://ppa.launchpad.net/qameta/allure/ubuntu xenial InRelease                                          
Hit:16 http://ppa.launchpad.net/ubuntu-desktop/ubuntu-make/ubuntu xenial InRelease
Hit:17 https://packagecloud.io/slacktechnologies/slack/debian jessie InRelease
Hit:18 http://ppa.launchpad.net/ubuntu-toolchain-r/test/ubuntu xenial InRelease
Hit:19 http://ppa.launchpad.net/viktor-krivak/pycharm/ubuntu xenial InRelease  
Hit:20 http://ppa.launchpad.net/webupd8team/java/ubuntu xenial InRelease       
Fetched 325 kB in 5s (63.6 kB/s)                   
Reading package lists... Done
Building dependency tree       
Reading state information... Done
All packages are up to date.
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Calculating upgrade... Done
The following packages were automatically installed and are no longer required:
  libllvm5.0 linux-headers-4.13.0-41 linux-headers-4.13.0-41-generic linux-headers-4.13.0-43 linux-headers-4.13.0-43-generic
  linux-headers-4.13.0-45 linux-headers-4.13.0-45-generic linux-headers-4.15.0-29 linux-headers-4.15.0-29-generic
  linux-image-4.13.0-41-generic linux-image-4.13.0-43-generic linux-image-4.13.0-45-generic linux-image-4.15.0-29-generic
  linux-image-extra-4.13.0-41-generic linux-image-extra-4.13.0-43-generic linux-image-extra-4.13.0-45-generic linux-modules-4.15.0-29-generic
  linux-modules-extra-4.15.0-29-generic
Use 'sudo apt autoremove' to remove them.
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Reading package lists... Done
Building dependency tree       
Reading state information... Done
avahi-daemon is already the newest version (0.6.32~rc+dfsg-1ubuntu2.3).
curl is already the newest version (7.47.0-1ubuntu2.12).
The following packages were automatically installed and are no longer required:
  libllvm5.0 linux-headers-4.13.0-41 linux-headers-4.13.0-41-generic linux-headers-4.13.0-43 linux-headers-4.13.0-43-generic
  linux-headers-4.13.0-45 linux-headers-4.13.0-45-generic linux-headers-4.15.0-29 linux-headers-4.15.0-29-generic
  linux-image-4.13.0-41-generic linux-image-4.13.0-43-generic linux-image-4.13.0-45-generic linux-image-4.15.0-29-generic
  linux-image-extra-4.13.0-41-generic linux-image-extra-4.13.0-43-generic linux-image-extra-4.13.0-45-generic linux-modules-4.15.0-29-generic
  linux-modules-extra-4.15.0-29-generic
Use 'sudo apt autoremove' to remove them.
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Reading package lists... Done
Building dependency tree       
Reading state information... Done
autoconf is already the newest version (2.69-9).
automake is already the newest version (1:1.15-4ubuntu1).
build-essential is already the newest version (12.1ubuntu2).
g++ is already the newest version (4:5.3.1-1ubuntu1).
make is already the newest version (4.1-6).
cmake is already the newest version (3.5.1-1ubuntu3).
uuid-dev is already the newest version (2.27.1-6ubuntu3.6).
The following packages were automatically installed and are no longer required:
  libllvm5.0 linux-headers-4.13.0-41 linux-headers-4.13.0-41-generic linux-headers-4.13.0-43 linux-headers-4.13.0-43-generic
  linux-headers-4.13.0-45 linux-headers-4.13.0-45-generic linux-headers-4.15.0-29 linux-headers-4.15.0-29-generic
  linux-image-4.13.0-41-generic linux-image-4.13.0-43-generic linux-image-4.13.0-45-generic linux-image-4.15.0-29-generic
  linux-image-extra-4.13.0-41-generic linux-image-extra-4.13.0-43-generic linux-image-extra-4.13.0-45-generic linux-modules-4.15.0-29-generic
  linux-modules-extra-4.15.0-29-generic
Use 'sudo apt autoremove' to remove them.
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Note, selecting 'zlib1g-dev' instead of 'libz-dev'
libboost-dev is already the newest version (1.58.0.1ubuntu1).
libtool is already the newest version (2.4.6-0.1).
libboost-system-dev is already the newest version (1.58.0.1ubuntu1).
libboost-thread-dev is already the newest version (1.58.0.1ubuntu1).
libpq-dev is already the newest version (9.5.14-0ubuntu0.16.04).
libssl-dev is already the newest version (1.0.2g-1ubuntu4.15).
zlib1g-dev is already the newest version (1:1.2.8.dfsg-2ubuntu4.1).
The following packages were automatically installed and are no longer required:
  libllvm5.0 linux-headers-4.13.0-41 linux-headers-4.13.0-41-generic linux-headers-4.13.0-43 linux-headers-4.13.0-43-generic
  linux-headers-4.13.0-45 linux-headers-4.13.0-45-generic linux-headers-4.15.0-29 linux-headers-4.15.0-29-generic
  linux-image-4.13.0-41-generic linux-image-4.13.0-43-generic linux-image-4.13.0-45-generic linux-image-4.15.0-29-generic
  linux-image-extra-4.13.0-41-generic linux-image-extra-4.13.0-43-generic linux-image-extra-4.13.0-45-generic linux-modules-4.15.0-29-generic
  linux-modules-extra-4.15.0-29-generic
Use 'sudo apt autoremove' to remove them.
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Reading package lists... Done
Building dependency tree       
Reading state information... Done
python-dbus is already the newest version (1.2.0-3).
python3-dev is already the newest version (3.5.1-3).
python-dev is already the newest version (2.7.12-1~16.04).
python3-pip is already the newest version (8.1.1-2ubuntu0.4).
The following packages were automatically installed and are no longer required:
  libllvm5.0 linux-headers-4.13.0-41 linux-headers-4.13.0-41-generic linux-headers-4.13.0-43 linux-headers-4.13.0-43-generic
  linux-headers-4.13.0-45 linux-headers-4.13.0-45-generic linux-headers-4.15.0-29 linux-headers-4.15.0-29-generic
  linux-image-4.13.0-41-generic linux-image-4.13.0-43-generic linux-image-4.13.0-45-generic linux-image-4.15.0-29-generic
  linux-image-extra-4.13.0-41-generic linux-image-extra-4.13.0-43-generic linux-image-extra-4.13.0-45-generic linux-modules-4.15.0-29-generic
  linux-modules-extra-4.15.0-29-generic
Use 'sudo apt autoremove' to remove them.
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Reading package lists... Done
Building dependency tree       
Reading state information... Done
libsqlite3-dev is already the newest version (3.11.0-1ubuntu1.1).
sqlite3 is already the newest version (3.11.0-1ubuntu1.1).
The following packages were automatically installed and are no longer required:
  libllvm5.0 linux-headers-4.13.0-41 linux-headers-4.13.0-41-generic linux-headers-4.13.0-43 linux-headers-4.13.0-43-generic
  linux-headers-4.13.0-45 linux-headers-4.13.0-45-generic linux-headers-4.15.0-29 linux-headers-4.15.0-29-generic
  linux-image-4.13.0-41-generic linux-image-4.13.0-43-generic linux-image-4.13.0-45-generic linux-image-4.15.0-29-generic
  linux-image-extra-4.13.0-41-generic linux-image-extra-4.13.0-43-generic linux-image-extra-4.13.0-45-generic linux-modules-4.15.0-29-generic
  linux-modules-extra-4.15.0-29-generic
Use 'sudo apt autoremove' to remove them.
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Cloning into '/tmp/foglamp-north-kafka'...
remote: Enumerating objects: 87, done.
remote: Counting objects: 100% (87/87), done.
remote: Compressing objects: 100% (55/55), done.
remote: Total 87 (delta 31), reused 57 (delta 19), pack-reused 0
Unpacking objects: 100% (87/87), done.
Checking connectivity... done.
-- The C compiler identification is GNU 5.5.0
-- The CXX compiler identification is GNU 5.5.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- No options set.
   +Using found FOGLAMP_ROOT /home/foglamp/Development/FogLAMP
-- Using FOGLAMP_ROOT for includes
   +/home/foglamp/Development/FogLAMP/C/common/include
   +/home/foglamp/Development/FogLAMP/C/services/common/include
   +/home/foglamp/Development/FogLAMP/C/plugins/filter/common/include
   +/home/foglamp/Development/FogLAMP/C/thirdparty/rapidjson/include
-- Using FOGLAMP_ROOT for libs 
   +/home/foglamp/Development/FogLAMP/cmake_build/C/lib
-- Installing Kafka in /home/foglamp/Development/FogLAMP/plugins/north/Kafka
-- Configuring done
-- Generating done
-- Build files have been written to: /tmp/foglamp-north-kafka/build
[ 25%] Generating version header
Scanning dependencies of target Kafka
[ 50%] Building CXX object CMakeFiles/Kafka.dir/kafka.o
[ 75%] Building CXX object CMakeFiles/Kafka.dir/plugin.o
[100%] Linking CXX shared library libKafka.so
[100%] Built target Kafka
[100%] Built target Kafka
Install the project...
-- Install configuration: ""
-- Installing: /home/foglamp/Development/FogLAMP/plugins/north/Kafka/libKafka.so.1
-- Installing: /home/foglamp/Development/FogLAMP/plugins/north/Kafka/libKafka.so
-- Set runtime path of "/home/foglamp/Development/FogLAMP/plugins/north/Kafka/libKafka.so.1" to ""
/home/foglamp/Development/FogLAMP
Kafka plugin is installed.
>>> Make sure south CoAP plugin service is running 
 & listening on specified host and port 

Total Statistics:

Start Time: 2019-03-15 18:35:00.068385
End Time:   2019-03-15 18:35:00.089264

Total Messages Transferred: 1
Total Bytes Transferred:    288

Total Iterations: 1
Total Messages per Iteration: 1.0
Total Bytes per Iteration:    288.0

Min messages/second: 47.895014129029164
Max messages/second: 47.895014129029164
Avg messages/second: 47.895014129029164

Min Bytes/second: 13793.7640691604
Max Bytes/second: 13793.7640691604
Avg Bytes/second: 13793.7640691604
/home/foglamp/Development/FogLAMP

PASSED
===== 1 passed in 110.58 seconds ====
```

Reply:
![reading-csv](https://user-images.githubusercontent.com/1974003/54435569-cdfa4000-4756-11e9-8302-73842a05b99f.png)